pages/_app.js

```diff
@@ -28,6 +29,13 @@ export default function App({ Component, pageProps }) {
       );
     }
   }
+
+  function handleAddActivity(newActivity) {
+    setActivitiesInfo((activitiesInfo) => [
+      { id: uid(), ...newActivity },
+      ...activitiesInfo,
+    ]);
+  }
```

Comment:
Great job ensuring that the state update is based on the most recent state! This attention to detail helps prevent potential bugs and makes the application more robust. Nicely done!